src/main/java/uk/gov/hmcts/reform/hmc/api/controllers/HearingsController.java

```diff
@@ -325,6 +328,44 @@ public ResponseEntity<Object> getFutureHearings(
         }
     }
 
+    /**
+     * End point to create hearing details for a case.
+     *
+     * @return success response, and initiates async process to create hearings
+     *     passed.
+     * @header authorization, user authorization token.
+     * @header serviceAuthorization, S2S authorization token.
+     * @RequestBody caseDetails to take all the hearings belongs to case.
+     */
+    @PostMapping(path = "/automated-hearing")
+    @ApiOperation("create automated hearing")
+    @ApiResponses(
+            value = {
+                @ApiResponse(code = 200, message = "Initiated Async Create hearings successfully"),
+                @ApiResponse(code = 400, message = "Bad Request")
+            })
+    public ResponseEntity<Object> createAutomatedHearings(
+            @RequestHeader(AUTHORIZATION) String authorization,
+            @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthorization,
+            @RequestBody final CaseData caseData) {
```

Comment:
Is it possible to make this more specific to hearing request than case data? Please let me know your view if the casedata is really needed here.